java/arcs/core/storage/handle/CollectionImpl.kt

```diff
@@ -90,6 +95,9 @@ class CollectionImpl<T : Referencable>(
         log.debug { "Storing: $entity" }
         @Suppress("GoodTime") // use Instant
         entity.creationTimestamp = requireNotNull(time).currentTimeMillis
+        require(entity !is RawEntity || schema == null || schema.refinement(entity)) {
```

Comment:
Optional (mostly for education purposes): you can use the `?.` operator here:

```kotlin
require(entity !is RawEntity || schema?.refinement(entity) != false) {
```

The right-side of the `||` translates to "boolean? != false" (nullable boolean) which is why the `!= false` is required, because `boolean?` can be `true`, `false`, or `null`

Reply:
That's neat but I think in this case its less readable and will leave it.